framework/include/constraints/ADMortarConstraint.h

```diff
@@ -73,3 +74,39 @@ class ADMortarConstraint : public MortarConstraintBase
   /// The primal solution gradient on the primary side
   const ADVariableGradient & _grad_u_primary;
 };
+
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+template <typename Variables, typename DualNumbers>
+void
+ADMortarConstraint::trimInteriorNodeDerivatives(
+    const std::map<unsigned int, unsigned int> & domain_ip_lowerd_map,
+    const Variables & moose_vars,
+    DualNumbers & dual_numbers,
+    const bool is_secondary)
+{
+  // Remove interior node variable's derivatives from AD objects.
+  for (const auto dof_index :
+       (is_secondary ? make_range(_test_secondary.size()) : make_range(_test_primary.size())))
+    if (!domain_ip_lowerd_map.count(dof_index))
+    {
+      for (const auto * const moose_var : moose_vars)
+      {
+        // It's valid for a user to pass a container that represents a LIBMESH_DIM vector of
+        // component variables for which one or two of the variables may be null depending on the
+        // mesh dimension in the simulation
+        if (!moose_var)
+          continue;
+
+        mooseAssert(moose_var->isNodal(),
+                    "Trimming of interior node's derivatives is only supported for Lagrange "
+                    "elements in mortar constraints");
+
+        const auto remove_derivative_index = is_secondary
+                                                 ? moose_var->dofIndices()[dof_index]
+                                                 : moose_var->dofIndicesNeighbor()[dof_index];
+        for (auto & dual_number : dual_numbers)
+          trimDerivative(remove_derivative_index, dual_number);
```

Comment:
Now we look for all the variable indices in the dual numbers. I think this makes it more general.